File: 3426-minimum-number-of-chairs-in-a-waiting-room/minimum-number-of-chairs-in-a-waiting-room.py
class Solution(object):
    def minimumChairs(self, s):
        """
        :type s: str
        :rtype: int
        """
        current_people = 0
        max_people = 0
    
        for ch in s:
            if ch == 'E':  
                current_people += 1
            else:  
                current_people -= 1
        
    
            max_people = max(max_people, current_people)
    
        return max_people

File: 3426-minimum-number-of-chairs-in-a-waiting-room/minimum-number-of-chairs-in-a-waiting-room.java
class Solution {
    public int minimumChairs(String s) {
        int chairs=0;
        int count=0;
        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            if(ch=='E'){
                count++;
                if(count>chairs){
                    chairs++;
                }
            }
            else{
                count--;
            }
        }
        return chairs;
        
    }
}